src/applications/personalization/profile-2/components/MobileMenuTrigger.jsx

```diff
@@ -0,0 +1,133 @@
+import React, { useEffect, useState, useRef } from 'react';
+import classnames from 'classnames';
+import { connect } from 'react-redux';
+
+import {
+  closeSideNav as closeSideNavAction,
+  openSideNav as openSideNavAction,
+  pinMenuTrigger as pinMenuTriggerAction,
+  unpinMenuTrigger as unpinMenuTriggerAction,
+} from '../actions';
+
+const MobileMenuTrigger = ({
+  closeSideNav,
+  focusTriggerButton,
+  openSideNav,
+  isMenuTriggerPinned,
+  pinMenuTrigger,
+  unpinMenuTrigger,
+}) => {
+  const button = useRef(null);
+  const placeholder = useRef(null);
+
+  const [mobileBreakpoint] = useState(767);
+  const [isMobile, setIsMobile] = useState(false);
+  const [triggerPosition, setTriggerPosition] = useState(null);
+  const [triggerHeight, setTriggerHeight] = useState(null);
+
+  // init some local state
+  useEffect(
+    () => {
+      setTriggerHeight(window.getComputedStyle(button.current).height);
+      setTriggerPosition(placeholder.current.offsetTop);
+      setIsMobile(window.innerWidth <= mobileBreakpoint);
+    },
+    [mobileBreakpoint],
+  );
+
+  useEffect(() => {
+    const handleResize = () => {
+      const mobile = window.innerWidth <= mobileBreakpoint;
+      if (!mobile) {
+        if (isMobile) {
+          setIsMobile(false);
+          closeSideNav();
+        }
+        return;
+      }
+      // The position of the trigger might move up or down as the window width
+      // changes and elements above it compress and take up more vertical space
+      // so we need to reset it on window resize
+      setTriggerPosition(placeholder.current.offsetTop);
+      if (!isMobile) {
+        setIsMobile(true);
+      }
+    };
+    // pin or unpin the mobile trigger
+    const handleScroll = () => {
+      if (!isMobile) return;
+      if (window.pageYOffset >= triggerPosition && !isMenuTriggerPinned) {
```

Comment:
Can we assign both `window.pageYOffset >= triggerPosition && !isMenuTriggerPinned` and `window.pageYOffset < triggerPosition && isMenuTriggerPinned` to descriptive `consts` so it's more readable to developers who don't necessarily know exactly what's going on?